doc/examples/workflow_creation.py

```diff
@@ -0,0 +1,127 @@
+"""
+============================================================
+Creating a new workflow.
+============================================================
+
+A workflow is a series of dipy operations with fixed inputs and outputs
+that is callable via commandline.
+
+For example, after installing dipy, you can call anywhere from your command line:
+$ dipy_nlmeans t1.nii.gz t1_denoised.nii.gz
+"""
+
+"""
+First create your workflow. Usually this would be in its own python file in
+the ../dipy/workflows directory.
+"""
+
+import shutil
+
+"""
+``shutil`` Will be used for sample file manipulation.
+"""
+
+from dipy.workflows.workflow import Workflow
+
+"""
+``Workflow`` is the base class that will be extended to create our workflow.
+"""
+
+
+class AppendTextFlow(Workflow):
+
+    def run(self, input_files, text_to_append='dipy', out_dir='',
+            out_file='append.txt'):
+        """
+        Parameters
+        ----------
+        input_files : string
+            Path to the input files. This path may contain wildcards to
+            process multiple inputs at once.
+
+        text_to_append : string, optional
+            Text that will be appended to the file. (default 'dipy')
+
+        out_dir : string, optional
+            Where the resulting file will be saved. (default '')
+
+        out_file : string, optional
+            Name of the result file to be saved. (default 'append.txt')
+        """
+
+        """
+        ``AppendTextFlow`` is the name of our workflow. Note that it needs
+        to extend Workflow for everything to work properly. It will append
+        text to a file.
+
+        It is mandatory to have out_dir as a parameter. It is also mandatory
+        to put 'out_' in front of every parameter that is going to be an
+        output. Lastly, all out_ params needs to be at the end of the params
+        list.
+
+        The class docstring part is very important, you need to document
+        every parameter as they will be used with inspection to build the
+        command line argument parser.
+        """
+
+        io_it = self.get_io_iterator()
+
+        for in_file, out_file in io_it:
```

Comment:
This `out_file` variable shadows the `out_file` parameter. Is this expected? If so, I'm guessing the parameter `out_file` is used when you call `self.get_io_iterator()` via introspection?